src/app/zap_cluster_list.py

```diff
@@ -21,18 +21,14 @@ def get_cluster_sources(clusters: typing.Set[str],
     cluster_sources: typing.Set[str] = set()
 
     for cluster in clusters:
-        if cluster not in source_map:
-            raise ValueError("Unhandled %s cluster: %s"
-                             " (hint: add to src/app/zap_cluster_list.json)" % (side, cluster))
-
-        cluster_sources.update(source_map[cluster])
+        if cluster in source_map:
```

Comment:
we seem to lose an exception here and silently ignore a cluster that is not in a source map.

Silently skipping things will be hard to debug.

Reply:
we need a warning at a minimum or a flag that makes it "these are known non-sdk clusters, skip" (which I imagine is what the old flag was supposed to do)

Reply:
Hi @andy31415 . In my opinion the exception is redundant:
- if vendor adds a custom cluster, they must modify the .json file, even if they source custom cluster separately (for example they don't want to put the implementation in `"${_app_root}/clusters/${cluster}/${cluster}.cpp"`)
- if new common cluster is created, but not added to `zap_cluster_list.json`, build will fail as files are not sourced. I aggree that it would be worth adding a note, but I'm not sure if Build GN can parse python's stderr output in `exec_script` ?
 
Alternatively I could convert the external_clusters flag from list to just boolean flag, so that it should be only set if we use external clusters, but without a burden of listing all of them every time?